### ISSUE TITLE
Lower performance on Kinetics using model zoo TDN-ResNet50 checkpoint

### ISSUE THREAD

**Issue by caidonkey** [MCG-NJU/TDN#2]
I followed your Testing instructions (3-crops, 10-clips) and evaluated model zoo TDN-ResNet50 checkpoint on Kinetics.
The only difference is that I use extracted frames as you process Something-V1&V2 instead of raw video inputs you process Kinetics.
I ran the Testing twice and got 75.05/91.95  and 75.10/91.97 respectively.
I know there is some dataset differences but it should not bring such a big performance gap.
Would you kindly check if the Testing setting is the one you obtain the reported performance?

**Comment by caidonkey** [comment]
Besides, there is a small bug in main.py Line 220:

>  prec1, prec5, val_loss = validate(val_loader, model, criterion, epoch, logger)

The parameter 'epoch' here should be removed.

**Comment by yztongzhan** [comment]
Hi @caidonkey , Thanks for your comments.  I think you might run this scripts for one time, which only does one clip and three crop test.
```
CUDA_VISIBLE_DEVICES=0 python3 test_models_three_crops.py  kinetics \
--archs='resnet50' --weights <your_checkpoint_path>  --test_segments=8 \
--test_crops=3 --batch_size=16 --full_res --gpus 0 --output_dir <your_pkl_path>  \
-j 4 --clip_index <your_clip_index>
```
You need run this script for ten times and change the <your_clip_index> from 0-9 then you will ensemble 30 pkl files with `pkl_to_results.py`

**Comment by yztongzhan** [comment]
> The only difference is that I use extracted frames as you process Something-V1&V2 instead of raw video inputs you process Kinetics.

We use the raw video dataset for online training and inference, the shorter side of each video is 320p, which might result in lower performance if you extract frames with lower resoultion
If you have more questions, please free to comments or email me tongzhan@smail.nju.edu.cn

**Comment by caidonkey** [comment]
> Hi @caidonkey , Thanks for your comments. I think you might run this scripts for one time, which only does one clip and three crop test.
> 
> ```
> CUDA_VISIBLE_DEVICES=0 python3 test_models_three_crops.py  kinetics \
> --archs='resnet50' --weights <your_checkpoint_path>  --test_segments=8 \
> --test_crops=3 --batch_size=16 --full_res --gpus 0 --output_dir <your_pkl_path>  \
> -j 4 --clip_index <your_clip_index>
> ```
> 
> You need run this script for ten times and change the <your_clip_index> from 0-9 then you will ensemble 30 pkl files with `pkl_to_results.py`

No, I have ran the test from 0-9 index and emsembled the results as you mentioned. 
The only difference is that I use 340x256 extracted frames. I'll check later with raw video inputs.

Thanks.

**Comment by yztongzhan** [comment]
Hi,  @caidonkey ! you can also check this [Data Benchmark](https://github.com/open-mmlab/mmaction2/blob/master/configs/recognition/slowonly/README.md#kinetics-400-data-benchmark) done by [MMaction2](https://github.com/open-mmlab/mmaction2).

**Comment by caidonkey** [comment]
> Hi, @caidonkey ! you can also check this [Data Benchmark](https://github.com/open-mmlab/mmaction2/blob/master/configs/recognition/slowonly/README.md#kinetics-400-data-benchmark) done by [MMaction2](https://github.com/open-mmlab/mmaction2).

Thanks for your kind remind.

I have used MMAction and MMAction2 to conduct extensive experiemnts, but with only 4 GPUs. The performance seemed reasonable. (340x256 extracted frames;  224x224 input)

Just considering the dataset difference, using MMaction2, under config "tsm_r50_dense_1x1x8
_100e_kinetics400_rgb", model zoo reported 72.9/90.44 and we obtained 72.74/90.37.

Recently, I also evaluated TAM (https://github.com/liu-zhy/temporal-adaptive-module) and obtained 75.25/91.99 on our validation set (76.1/92.3 reported). But when we trained using the released config setting, we could obtain 75.59/92.04. I think the performance gap by data is also reasonable.

**Comment by MartinXM** [comment]
Hi, @yztongzhan nice work!  Can you share your version of kinetics400 with me? My current version is similar to @caidonkey , with short side equals 256. My email address is marquezxm@gmail.com. Many thanks !

**Comment by Brasiletto** [comment]
您好！我在训练后得到最好的模型为73.13/91.16，但是测试效果一直接近0不知道什么原因，您方便把kinetics测试的指令及代码发给我参考一下吗？谢谢！

**Comment by Brasiletto** [comment]
> 我按照您的测试说明（3 次裁剪、10 次剪辑）并在 Kinetics 上评估了模型动物园 TDN-ResNet50 检查点。
> 唯一的区别是我在处理Something-V1&V2时使用提取的帧，而不是处理Kinetics的原始视频输入。
> 我运行了两次测试，分别得到了 75.05/91.95 和 75.10/91.97。
> 我知道有一些数据集差异，但它不应该带来如此大的性能差距。
> 请您检查一下测试设置是否是您获得报告性能的设置？

你好！我在训练后得到最好的模型为73.13/91.16，但测试效果一直接近00不知道什么原因，你可以把动力学测试的指令和代码知道我看看一下吗？谢谢！

**Comment by Brasiletto** [comment]
> 我按照您的测试说明（3 次裁剪、10 次剪辑）并在 Kinetics 上评估了模型动物园 TDN-ResNet50 检查点。
> 唯一的区别是我在处理Something-V1&V2时使用提取的帧，而不是处理Kinetics的原始视频输入。
> 我运行了两次测试，分别得到了 75.05/91.95 和 75.10/91.97。
> 我知道有一些数据集差异，但它不应该带来如此大的性能差距。
> 请您检查一下测试设置是否是您获得报告性能的设置？

你好！我在训练后得到最好的模型为73.13/91.16，但测试效果一直接近0不知道什么原因，可以把动力学测试的指令和代码知道我看看一下吗？谢谢！

**Comment by Brasiletto** [comment]
> I followed your Testing instructions (3-crops, 10-clips) and evaluated model zoo TDN-ResNet50 checkpoint on Kinetics.
> The only difference is that I use extracted frames as you process Something-V1&V2 instead of raw video inputs you process Kinetics.
> I ran the Testing twice and got 75.05/91.95 and 75.10/91.97 respectively.
> I know there is some dataset differences but it should not bring such a big performance gap.
> Would you kindly check if the Testing setting is the one you obtain the reported performance?

Hi! The best model I got after training is 73.13/91.16, but the test result is always close to 0. I don’t know why, can you send me the instructions and code for the kinetics test? Thanks!
